### PR TITLE
Refactor Resolve

### PR DESCRIPTION
**Pull request by oberien**

### CODE REVIEW

File: src/resolve/providers.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
+use std::io;
+use std::path::{PathBuf, Path, Component};
+
+use url::{Url, Host};
+
+use resolve::{Resource, ResourceBuilder, Source, InnerSource, Include, Command};
+
+// We Java now
```
Comment (197g) [inline src/resolve/providers.rs]: Rename `ResourceProvider => Server`, `ResourceProviderProvider => ServerProvider`

**Comment by oberien** [comment]
@HeroicKatora Do you plan on reviewing this or can I merge it without your evaluation? (If you'd like to but don't have time right now that's fine, there's plenty of other stuff that needs to be done :-) )

**Comment by 197g** [comment]
I'll have a more thorough look on the weekend if you want but the most important aspects of resolution should have been covered by the simple test anyways. Moving the host based resolution into its own type and `impl` and removing the indirection through the hashmap is likely better for now until there is a use case for changing that dynamically.

File: src/resolve/mod.rs [hunk 2]
```diff
@@ -257,8 +196,8 @@ mod tests {
     fn domain_resolves() {
```
Comment (197g) [inline src/resolve/mod.rs:146]: This test needs to be reworked. 

https://github.com/oberien/pundoc/blob/d193191e5178075734fd88d88ff3246fb43dc717/src/resolve/mod.rs#L197-L203

Remove the `struct` and `add_provider` call then the builtin hardcoded mechanism should take over.

Comment (197g) [reply src/resolve/mod.rs:146]: Without regards to the review content, is it possible to tag multiple and existing lines in a review? That would have made the comment more helpful.

Comment (oberien) [reply src/resolve/mod.rs:146]: Not for a review comment directly, but within the review comment you can use [this](https://github.community/t5/How-to-use-Git-and-GitHub/PR-multiline-review/m-p/6779#M2261).

**Comment by oberien** [comment]
@HeroicKatora I went back to the roots and tried to find out the question we are trying to solve with this PR. Landing at #3 I completely refactored this code again to make it simpler and more straightforward, less overengineered.
Could you please give this version another review?

**Pull request merged**